### PR TITLE
fix: ignore baseline if path was not relative in source

### PR DESCRIPTION
**Pull request by raffis**
### Description:
There is currently a bug with the baseline ignore.

If gitleaks is not executed in the current directory and --source is used to specify a custom path, the baseline file is not ignored correctly because the relative file paths do not match. That check is done in `Detect()`.

This will currently report secrets in the baseline file, however the baseline file should be excluded.
```
gitleaks detect --report-format json --verbose --log-level debug --baseline-path ~/foo/.gitleaks.baseline.json --source ~/foo
```

Basically as for now the baseline only works if gitleaks is executed directly in the target repository and the baseline path is set relative to it.
This pr fixes that bug so the baseline path is resolved correctly and can be compared during fragments check no matter what directory.

### Checklist:

* [x] Does your PR pass tests?
* [x] Have you written new tests for your changes?
* [x] Have you lint your code locally prior to submission?

### CODE REVIEW

File: detect/detect_test.go [hunk 1]
```diff
@@ -318,6 +319,15 @@ func TestDetect(t *testing.T) {
 			},
 			expectedFindings: []report.Finding{},
 		},
+		{
+			cfgName:      "path_only",
+			baselinePath: ".baseline.json",
+			fragment: Fragment{
+				Raw:      `const Discord_Public_Key = "e7322523fb86ed64c836a979cf8465fbd436378c653c1db38f9ae87bc62a6fd5"`,
```
[the remark this answers is not in the document]
Comment (raffis) [reply detect/detect_test.go:326]: #jit_ignore_fp

**Comment by zricethezav** [comment]
@raffis thank you for this!
